bundles/org.openhab.binding.mielecloud/src/main/java/org/openhab/binding/mielecloud/internal/handler/AbstractMieleThingHandler.java

```diff
@@ -0,0 +1,291 @@
+
+    /**
+     * Creates a new {@link AbstractMieleThingHandler}.
+     * 
+     * @param thing The thing to handle.
+     */
+    public AbstractMieleThingHandler(Thing thing) {
+        super(thing);
+        this.actionFetcher = new ActionStateFetcher(this::getWebservice, scheduler);
+    }
+
+    private Optional<MieleBridgeHandler> getMieleBridgeHandler() {
+        Bridge bridge = getBridge();
+        if (bridge == null) {
+            return Optional.empty();
+        }
+
+        BridgeHandler handler = bridge.getHandler();
+        if (handler == null || !(handler instanceof MieleBridgeHandler)) {
+            return Optional.empty();
+        }
+
+        return Optional.of((MieleBridgeHandler) handler);
+    }
+
+    protected MieleWebservice getWebservice() {
+        return getMieleBridgeHandler().map(MieleBridgeHandler::getWebservice)
+                .orElse(UnavailableMieleWebservice.INSTANCE);
+    }
+
+    @Override
+    public void initialize() {
+        getWebservice().dispatchDeviceState(getDeviceId());
+
+        // If no device state update was received so far, set the device to OFFLINE.
+        if (getThing().getStatus() == ThingStatus.INITIALIZING) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (RefreshType.REFRESH.equals(command)) {
+            updateDeviceState(new DeviceChannelState(latestDeviceState));
+            updateTransitionState(new TransitionChannelState(latestTransitionState));
+            updateActionState(new ActionsChannelState(latestActionsState));
+        }
+
+        switch (channelUID.getId()) {
+            case PROGRAM_START_STOP:
+                if (PROGRAM_STARTED.matches(command.toString())) {
+                    triggerProcessAction(START);
+                } else if (PROGRAM_STOPPED.matches(command.toString())) {
+                    triggerProcessAction(STOP);
+                }
+                break;
+
+            case PROGRAM_START_STOP_PAUSE:
+                if (PROGRAM_STARTED.matches(command.toString())) {
+                    triggerProcessAction(START);
+                } else if (PROGRAM_STOPPED.matches(command.toString())) {
+                    triggerProcessAction(STOP);
+                } else if (PROGRAM_PAUSED.matches(command.toString())) {
+                    triggerProcessAction(PAUSE);
+                }
+                break;
+
+            case LIGHT_SWITCH:
+                if (command instanceof OnOffType) {
+                    triggerLight(OnOffType.ON.equals(command));
+                }
+                break;
+
+            case POWER_ON_OFF:
+                if (POWER_ON.matches(command.toString()) || POWER_OFF.matches(command.toString())) {
+                    triggerPowerState(OnOffType.ON.equals(OnOffType.from(command.toString())));
```

Comment:
This replacement doesn't work. `command` is of type `StringType` (because the `POWER_ON_OFF` channel is a string channel with options `"on"` and `"off"`). Thus, we need to convert it here.

A note on why this is a `String` channel: Initially the binding was planned to have an additional channel `POWER_ON_OFF_STANDBY` that was not implemented for the contribution but may be added in the future. For consistency purposes we decided to make `POWER_ON_OFF` a `String` channel, too. Would you prefer to convert it to a `Switch` channel for now?

Reply:
I overlooked that it's of StringType. I'd keep it as StringType, otherwise you would break compatibility when changing it from Switch to StringType later.